dhall/src/Dhall/DirectoryTree.hs

```diff
@@ -409,57 +420,57 @@ instance Show FilesystemError where
-          \In contrast, the following expression is not allowed due to containing a           \n\
-          \❰Natural❱ field, which cannot be translated in this way:                           \n\
-          \                                                                                   \n\
-          \                                                                                   \n\
-          \    ┌───────────────────────┐                                                      \n\
-          \    │ { `example.txt` = 1 } │                                                      \n\
-          \    └───────────────────────┘                                                      \n\
-          \                                                                                   \n\
-          \                                                                                   \n\
-          \Note that key names cannot contain path separators:                                \n\
-          \                                                                                   \n\
-          \                                                                                   \n\
-          \    ┌─────────────────────────────────────┐                                        \n\
-          \    │ { `directory/example.txt` = \"ABC\" } │ Invalid: Key contains a forward slash\n\
-          \    └─────────────────────────────────────┘                                        \n\
-          \                                                                                   \n\
-          \                                                                                   \n\
-          \Instead, you need to refactor the expression to use nested records instead:        \n\
-          \                                                                                   \n\
-          \                                                                                   \n\
-          \    ┌───────────────────────────────────────────┐                                  \n\
-          \    │ { directory = { `example.txt` = \"ABC\" } } │                                \n\
-          \    └───────────────────────────────────────────┘                                  \n\
-          \                                                                                   \n\
-          \                                                                                   \n\
-          \You tried to translate the following expression to a directory tree:               \n\
-          \                                                                                   \n\
-          \" <> Util.insert unexpectedExpression <> "\n\
-          \                                                                                   \n\
+          Util._ERROR <> ": Not a valid directory tree expression                             \n\\
```

Comment:
The GHC documentation recommends appending a space so that cpp doesn't fall over, but that causes a different error. Adding a slash like this seems to work. Perhaps I should tell the GHC maintainers.

Reply:
Wait, something seems amiss here.  I was looking at this more closely and it does not seem to work in my local testing.  The following sample code I tested:

```haskell
module Main where

main :: IO ()
main = putStrLn "foo\n\\
                \bar\n\\
                \baz\n"
```

… gives me a lexical error:

```
Main.hs:4:25: error:
    lexical error in string/character literal at character '\n'
  |
4 | main = putStrLn "foo\n\\
  |                         ^
```

Yet clearly CI passed for your branch so I'm a bit confused.  My intuition, though, is that the double trailing slash should not work, though, since I thought that would get translated to a literal slash in the string.

Reply:
This works for me:

```haskell
{-# LANGUAGE CPP #-}

module Main where

main :: IO ()
main = putStrLn "foo\n\\
                \bar\n\\
                \baz\n"
```

Reply:
Ohhhh, I didn't realize that `CPP` changed the semantics of string literals.  Alright, that clears up my confusion.